api/graphql/mutations/user.js

```diff
@@ -2,14 +2,34 @@ import request from 'request'
 
 export function blockUser (userId, blockedUserId) {
   return BlockedUser.create(userId, blockedUserId)
-  .then(() => ({success: true}))
+  .then(() => ({ success: true }))
 }
 
 export async function unblockUser (userId, blockedUserId) {
   const blockedUser = await BlockedUser.find(userId, blockedUserId)
-  if (!blockedUser) throw new Error("user is not blocked")
+  if (!blockedUser) throw new Error('user is not blocked')
   return blockedUser.destroy()
-  .then(() => ({success: true}))
+  .then(() => ( {success: true }))
+}
+
+export async function deactivateUser ({ userId, sessionId }) {
+  const user = await User.find(userId)
+  await user.deactivate(sessionId)
+  return { success: true }
+}
+
+export async function reactivateUser ({ userId }) {
+  const user = await User.find(userId, {}, false)
+  await user.reactivate()
+  return { success: true }
+}
+
+export async function deleteUser ({ userId, sessionId }) {
```

Comment:
the permission check should be that the mutation only allows for deleting the current user, but this function can take a userId for potential other use in the future